src/heartbeat/listener.js

```diff
@@ -16,40 +16,72 @@ const utils = require('./utils');
 const configModule = require('../config/config');
 const collectorStatus = require('../constants').collectorStatus;
 const sanitize = require('../utils/commonUtils').sanitize;
+const repeater = require('../repeater/repeater');
+
+let pausedAfterHeartbeatError = false;
+
+function onError(err) {
+  try {
+    debug('heartbeat error %O', err);
+    logger.error('heartbeat error: ', err.message);
+
+    // Pause all the repeater generators until successful heartbeat.
+    repeater.pauseGenerators();
+    pausedAfterHeartbeatError = true;
+    return err; // returning the error here just for testability
+  } catch (err) {
+    /*
+     * Catching and handling here so it doesn't bubble up and get caught by
+     * heartbeat.js, which should only be catching errors from the POST.
+     */
```

Comment:
I don't think we need to try/catch in onError, since if it got here it already got to the catch in heartbeat.js.